benchmark/bench_bosehubbardmom12d2c.jl

```diff
@@ -0,0 +1,53 @@
+module BenchBoseHubbardMom1D2C
+
+using Rimu
+using BenchmarkTools
+
+
+suite = BenchmarkGroup()
+
+function run_lomc(vectype, H, addr, n_steps=5_000)
+    Nw_init = 10
+    Ntw = 2_000
+    dτ = 0.001
+    k = 1
+
+    c = vectype(Dict(addr => Nw_init), Ntw)
```

Comment:
I see that you are setting for the capacity the number of `targetwalkers`, which sounds reasonable. However, the logs show errors
```
julia> benchmarkpkg(Rimu)
PkgBenchmark: Running benchmarks...
PkgBenchmark: using benchmark tuning data in /Users/brand/git/Rimu/benchmark/tune.json
(1/2) benchmarking "bosehubbardmom12d2c"...
  (1/4) benchmarking "DVec small"...
  done (took 29.909243203 seconds)
  (2/4) benchmarking "DVec2 big"...
┌ Warning: maxlength nearly reached
│   len_local = 2629
│   maxlength = 2730
└ @ Rimu /Users/brand/git/Rimu/src/fciqmc.jl:322
┌ Error: maxlength exceeded
│   len_local = 3160
│   maxlength = 2730
└ @ Rimu /Users/brand/git/Rimu/src/fciqmc.jl:319
```
for both "DVec big" and "DVec2 big". This should be fixable by setting capacity higher. However, maybe we should think of something more clever, where we pass the expected average number of walker to the constructor of `DVec` and then the capacity is adjusted such that the underlying dictionary works at optimum filling factor.

Reply:
I tweaked the `dτ` a bit. Seems to work ok. I also tweaked the numbers of iterations to get approximately the same time for all benchmarks.